### PR TITLE
deprecate receiveUploadPermissions

### PR DESCRIPTION
**Pull request by jiteshdhamaniya**
## Description

This PR Solves https://github.com/WordPress/gutenberg/issues/33900. 

`receiveUploadPermissions` is not used anymore so this PR makes it deprecated and suggests to use `receiveUserPermission` instead. 

## How has this been tested?

- Find `receiveUploadPermissions` in all Gutenberg files, and found no reference, except declaration file. 
- Removed `receiveUploadPermissions` and tested site to confirm everything is working. 

## Screenshots 
Screenshot 2021-12-18 at 11 32 17 AM

Screenshot 2021-12-18 at 11 33 03 AM

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by swissspidy** [comment]
I thought the code can be safely removed altogether because it‘s not exposed publicly.

**Comment by Mamaduka** [comment]
> I thought the code can be safely removed altogether because it‘s not exposed publicly.

I think everything that's exported is public API>

File: packages/core-data/src/actions.js [hunk 1]
```diff
@@ -696,15 +696,17 @@ export const __experimentalSaveSpecifiedEntityEdits = (
...
+export function receiveUploadPermissions() {
+	deprecated( "wp.data.dispatch( 'core' ).receiveUploadPermissions", {
+		since: '5.9',
+	} );
+
 	return {
-		type: 'RECEIVE_USER_PERMISSION',
-		key: 'create/media',
-		isAllowed: hasUploadPermissions,
+		type: 'DO_NOTHING',
```
Comment (ntsekouras) [inline packages/core-data/src/actions.js]: We should still internally point to `receiveUserPermission` to keep backwards compatibility for now, no?

Comment (Mamaduka) [reply packages/core-data/src/actions.js]: I agree. I think we just need a deprecation message here.

Comment (jiteshdhamaniya) [reply packages/core-data/src/actions.js]: Something like this ? 

Screenshot 2021-12-21 at 9 23 32 PM

Comment (ntsekouras) [reply packages/core-data/src/actions.js]: Yes, but you should bring back the `hasUploadPermissions` param in the function's signature.

Comment (jiteshdhamaniya) [reply packages/core-data/src/actions.js]: oh yes. updated.

File: packages/core-data/src/actions.js [hunk 2]
```diff
@@ -696,16 +696,18 @@ export const __experimentalSaveSpecifiedEntityEdits = (
...
  * @return {Object} Action object.
  */
 export function receiveUploadPermissions( hasUploadPermissions ) {
-	return {
-		type: 'RECEIVE_USER_PERMISSION',
-		key: 'create/media',
-		isAllowed: hasUploadPermissions,
-	};
+	deprecated( "wp.data.dispatch( 'core' ).receiveUploadPermissions", {
+		since: '5.9',
```
Comment (ntsekouras) [inline packages/core-data/src/actions.js:707]:
```suggestion
		since: '5.9',
		alternative: 'receiveUserPermission',
```

**Review by ntsekouras — APPROVED** [review]
Thanks for your contribution @jiteshdhamaniya! 👍

**Comment by swissspidy** [comment]
> > I thought the code can be safely removed altogether because it‘s not exposed publicly.
> 
> I think everything that's exported is public API>

IIRC this function was never exported publicly outside of the package, so other packages could not use it. Hence the suggestion to completely remove this code

**Comment by Mamaduka** [comment]
Sorry if I misunderstand something. Developers could access this action using the example below, making it a public API:

```js
const { receiveUploadPermissions } = useDispatch( 'core' );
```

**Comment by swissspidy** [comment]
Ah, right, missed that. My bad.

**Pull request merged**